feature: created new component : spinning text

ibtihelbs opened the pull request.

ibelick commented on components/core/spinning-text.tsx:
```diff
@@ -0,0 +1,58 @@
+  children: ReactElement;
+  width?: number;
+  textStyle?: CSSProperties;
+  degree?: number;
```
can we add props:
- className
- duration

ibtihelbs replied:
I added both duration and className.

ibelick commented on components/core/spinning-text.tsx:
```diff
@@ -0,0 +1,58 @@
+}) => {
+  const textEl = children.props.children || children.props.content || '';
+  const toArr = typeof textEl === 'string' ? textEl.split('') : [];
+  console.log(toArr);
```
remove log

ibtihelbs replied:
done!

ibelick commented on components/core/spinning-text.tsx:
```diff
@@ -0,0 +1,58 @@
+
+  return (
+    <motion.div
+      className='border-3 relative flex items-center justify-center rounded-full border-current'
```
remove `border-3`

ibelick commented on components/core/spinning-text.tsx:
```diff
@@ -0,0 +1,58 @@
+        height: `${width}px`,
+        width: `${width}px`,
+      }}
+      animate={{ rotate: 360 }} // Rotate the entire circle
```
remove comments please (`// Rotate the entire circle` and others)

ibtihelbs replied:
done

ibtihelbs commented:
Changes are : Duration: Added a duration prop to control the speed of the rotation.
ClassName: Added a className prop for applying custom styles.
Text Extraction: Adjusted text extraction to handle both direct text and components with a content prop.
Default Values: Set default values for duration and className, with className defaulting to any class from the child component if provided.

ibtihelbs commented:
hope you check the new changes!

ibelick commented:
@ibtihelbs I'll take a look soon. First I need to better organize the motion-primitives components

ibelick commented:
Thanks for your contribution @ibtihelbs! Cf https://github.com/ibelick/motion-primitives/pull/68
